src/Poseidon/Chronicle.hs

```diff
@@ -0,0 +1,184 @@
+    , snapYamlDescription      :: Maybe String
+    , snapYamlChronicleVersion :: Version
+    , snapYamlLastModified     :: Day
+    , snapYamlPackages         :: S.Set PackageIteration
+    }
+    deriving (Show, Eq)
+
+instance FromJSON PoseidonPackageChronicle where
+    parseJSON = withObject "PoseidonYamlStruct" $ \v -> PoseidonPackageChronicle
+        <$> v .:  "title"
+        <*> v .:? "description"
+        <*> v .:  "chronicleVersion"
+        <*> v .:  "lastModified"
+        <*> v .:? "packages" .!= S.empty
+
+instance ToJSON PoseidonPackageChronicle where
+    toJSON x = object $ [
+        "title"            .= snapYamlTitle x,
+        "description"      .= snapYamlDescription x,
+        "chronicleVersion" .= snapYamlChronicleVersion x,
+        "lastModified"     .= snapYamlLastModified x] ++
+        if not $ null (snapYamlPackages x) then ["packages" .= snapYamlPackages x] else []
+
+-- | A data type to represent a package state
+data PackageIteration = PackageIteration
+    { pacStateTitle   :: String   -- ^ the title of the package
+    , pacStateVersion :: Version  -- ^ the version of the package
+    , pacStateCommit  :: String   -- ^ the hash of a relevant commit where a package can be accessed in this version
+    , pacStatePath    :: FilePath -- ^ the file path where the POSEIDON.yml file is stored
+    }
+    deriving (Show)
+
+instance Eq PackageIteration where
+    (PackageIteration t1 v1 _ _) == (PackageIteration t2 v2 _ _) = (t1 == t2) && (v1 == v2)
+
+instance Ord PackageIteration where
+    (PackageIteration t1 v1 _ _) `compare` (PackageIteration t2 v2 _ _) = (t1,v1) `compare` (t2,v2)
+
+instance FromJSON PackageIteration where
+    parseJSON = withObject "packages" $ \v -> PackageIteration
+        <$> v .: "title"
+        <*> v .: "version"
+        <*> v .: "commit"
+        <*> v .: "path"
+
+instance ToJSON PackageIteration where
+    toJSON x = object [
+          "title"   .= pacStateTitle x
+        , "version" .= pacStateVersion x
+        , "commit"  .= pacStateCommit x
+        , "path"    .= pacStatePath x
+        ]
+
+instance HasNameAndVersion PackageIteration where
+    getPacName (PackageIteration t _ _ _) = t
+    getPacVersion (PackageIteration _ v _ _) = Just v
+
+updateChronicle :: PoseidonPackageChronicle -> PoseidonPackageChronicle -> PoseidonPackageChronicle
+updateChronicle oldChronicle newChronicle =
+    let oldPackageSet = snapYamlPackages oldChronicle
+        newPackageSet = snapYamlPackages newChronicle
+        mergedPacSet = S.union oldPackageSet newPackageSet
+        oldChronicleVersion = snapYamlChronicleVersion oldChronicle
+    in PoseidonPackageChronicle {
+      snapYamlTitle            = snapYamlTitle oldChronicle
+    , snapYamlDescription      = snapYamlDescription oldChronicle
+    , snapYamlChronicleVersion = if mergedPacSet /= oldPackageSet
+                                 then updateThreeComponentVersion Minor oldChronicleVersion
```

Comment:
OK, and this decides on the minor version nr to increase. Are we going to manually change the major one at some point? Perhaps not something to decide on now.

Reply:
Right - I thought the automatic default for the GitHub action should be a change in the **minor** version component of the chronicle. An automatic update with `trident chronicle` means, that one or more packages were added, without changing or removing anything else. When we manually edit the chronicle (e.g. clean out unwanted package iterations), then we can manually set a new major version number.